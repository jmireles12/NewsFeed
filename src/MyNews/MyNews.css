.headlines {
    text-align: center;
}

.headline-list {
    list-style-type: none;
    padding: 0
}

.headline__li:nth-child(even) {
    background: #e9ecef;
}