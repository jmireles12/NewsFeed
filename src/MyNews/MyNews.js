import React, { Component } from 'react'
import ApiContext from '../ApiContext'
import Articles from '../Articles/Articles'
import Tabs from '../Tabs/Tabs'
import './MyNews.css'

export default class MyNews extends Component {

    static defaultProps = {
        match: {
            params: {}
        }
    }

    static contextType = ApiContext;


    render() {
        const { news=[] } = this.context
        return (
            <section>
                <Tabs />
                <ul className='headline-list'>
                    {news.map(article =>
                        <li className='headline__li' key={article.id}>
                            <Articles
                                id={article.id}
                                title={article.fields.headline}
                                body={article.fields.bodyText}
                                publication={article.webPublicationDate}
                                url={article.webUrl}
                                thumbnail={article.fields.thumbnail}
                            />
                        </li>)}
                </ul>
            </section>
        )
    }
}