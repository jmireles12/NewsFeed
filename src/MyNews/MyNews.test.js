import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MyNews from './MyNews'

describe(`MyNews component`, () => {
    it('renders the complete MyNews', () => {
        const wrapper = shallow(<MyNews />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})