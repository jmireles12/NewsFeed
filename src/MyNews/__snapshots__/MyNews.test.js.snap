// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyNews component renders the complete MyNews 1`] = `
<div>
  <Tabs
    match={
      Object {
        "params": Object {},
      }
    }
  />
  <ul
    className="headline-list"
  />
</div>
`;
